### PR TITLE
Replicator stop: avoid waiting for 10 seconds if replicator is already stopped

### PR DESCRIPTION
**Pull request by Nao-ris**

### CODE REVIEW

File: src/replicator.rs [hunk 1]
```diff
@@ -613,8 +613,11 @@ impl Replicator {
                 std::mem::transmute(&callback),
             );
 
-            CBLReplicator_Stop(self.get_ref());
-            let success = receiver.recv_timeout(Duration::from_secs(10)).is_ok();
+            let mut success = true;
+            if self.status().activity != ReplicatorActivityLevel::Stopped {
+                CBLReplicator_Stop(self.get_ref());
+                success = receiver.recv_timeout(Duration::from_secs(10)).is_ok();
```
Comment (n-eq) [inline src/replicator.rs:619]: Maybe we could have a public setter of this value to avoid potentially having to update the library

**Review by n-eq — APPROVED** [review]

File: src/replicator.rs (hunk 1 above)
Comment (Nao-ris) [reply src/replicator.rs:619]: On which value ?

Comment (yanbonnel) [reply src/replicator.rs:619]: `Durration::from_secs(10)` I think

Comment (Nao-ris) [reply src/replicator.rs]: Timeout added as parameter

**Pull request merged**